### DISCUSSION TITLE
Add operator from a vector

### DISCUSSION THREAD

**Discussion by mgallegoq** [netket/netket#1585] [Q&A]
I'd like to define a new operator as the tensor product of 2 states that I already have as $|\psi\rangle \langle \psi |$. I am having problems since this is not a Local operator.

**Comment by gcarleo** [comment]
Hi, this is not currently supported, since these are non local operators and they require special treatment when computing expected values etc. You would need to write your own operator to do that or maybe reuse some of the overlap functionalities developed by @alleSini99 https://github.com/netket/netket_fidelity

**Reply by mgallegoq** [reply]
Thank you! I'll check those.

**Comment by PhilipVinc** [comment]
The 'little code' way to do this would be to define a tensor ansatz of your two ansatze, and then use the infidelity operator from netket fidelity (and the associated theory discussed in the paper).

As Giuseppe says, it's not trivial to go from operators to projectors on states. I suggest reading the section of [that paper](https://arxiv.org/pdf/2305.14294) on infidelity minimisation which is close to what you want to do.

**Reply by mgallegoq** [reply]
Thank you! I'll consult the paper.

**Reply by alleSini99** [reply]
@mgallegoq if you need clarifications on how the code in `https://github.com/netket/netket_fidelity` works, do not hesitate to ask.

**Reply by mgallegoq** [reply]
Thank you!
